sympy/series/limits.py

```diff
@@ -375,7 +376,7 @@ def set_signs(expr):
             r = gruntz(e, z, z0, dir)
             if r is S.NaN or l is S.NaN:
                 raise PoleError()
-        except (PoleError, ValueError):
+        except (PoleError, ValueError, NotImplementedError):
```

Comment:
Nope, I don't think this should be done either. 

I know it might be confusing to understand as to when we want to raise a NotImplemented Error and when we want to return self when a limit can't be computed ... but there were decisions made on these things in the past (mainly during GSoc 2020, 2021 & 2022 when series modules was selected consecutively) ... and the whole point of inviting you on gitter was that you scroll through these tedious decisions whenever in doubt :)

I am sure we've discussed this on multiple times as this looks like a straightforward thing to do .

Reply:
1) It's not that we return self whenever SymPy can't compute a limit. Let's take Oscar's example here `limit(sin(x)**log(x), x, oo)`

This involves computing leadterm for `sin(1/x)**log(1/x)`, which can't be determined, so leadterm is `nan` 

<img width="717" alt="image" src="https://github.com/sympy/sympy/assets/87052487/a91b63f9-52bc-492e-909f-b3e28ed53078">

and hence there is nothing lacking on SymPy's whole limit pipeline here and in this case it makes sense to return `self`

Reply:
2) I am not fully sure and you might like to go through the discussions on gitter that we had but, we don't really want to catch NotImplemented Errors out of gruntz cause

a) gruntz is the last resolve, first we do all the leadterm bussiness and if nothing works we go to gruntz
b) now if you see the mrv_leadterm out of grutnz 
https://github.com/sympy/sympy/blob/de505ae8972198efe181afacdb643893b3511735/sympy/series/gruntz.py#L537-L542

This is what gruntz is dependent on in the end, now

i) calculate_series might not be able to calculate series because maybe series is not implemented
ii) even if series is implemented, we might not able to find leadterm cause that might not be implemented.

And we want to catch all these and make a note of what we don't have working in sympy and even the user would know what's missing currently.

Being the last resolve, I think it's better we make note of the errors rather than just return self and not have any clue as to what went wrong and the block above you edited ends up defeating the whole purpose I think 
```
except (PoleError, ValueError, NotImplementedError):
```

Reply:
ohk, I get your point, so maybe it is better to not cache NotImplementedError in such case and keep things as is, as the user will be prompted to add further assumptions for the symbol in question.